sdk/core/azure-core/src/io/body_stream.cpp

```diff
@@ -128,7 +128,10 @@ FileBodyStream::FileBodyStream(const std::string& filename)
   }
   catch (std::exception&)
   {
-    CloseHandle(fileHandle);
+    if (fileHandle != INVALID_HANDLE_VALUE)
```

Comment:
Do we really need this check or can we close a handle even if it is `INVALID_HANDLE_VALUE`?

Reply:
For the limited amount of time I researched, I haven't found a definitive answer, not closing it is safer - https://social.msdn.microsoft.com/Forums/en-US/ede8051a-ec1e-44b2-a42a-11332421d23e/is-quotclosehandle0quot-always-safe?forum=windowsgeneraldevelopmentissues